### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by joaquincamara**

### CODE REVIEW

File: toDoList/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/gorilla/mux"
+)
+
+type Event struct {
+	Id          int32
+	Title       string
+	Description string
+}
+
+var DB = make(map[int32]Event)
```
Comment (edu-transcarent) [inline toDoList/cmd/main.go:17]: why is this var public?

File: toDoList/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+func main() {
+	router := mux.NewRouter()
+	router.HandleFunc("/", Home)
+	router.HandleFunc("/api/v1/events", GetAllEvents).Methods("GET")
```
Comment (edu-transcarent) [inline toDoList/cmd/main.go:22]: missing testing.

File: toDoList/cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	router.HandleFunc("/api/v1/events", CreateEvent).Methods("POST")
+	router.HandleFunc("/api/v1/events", EditEvent).Methods("PUT")
+	router.HandleFunc("/api/v1/events", DeleteEvent).Methods("DELETE")
+
+	http.ListenAndServe(":8080", router)
+
+}
+
+func Home(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("Welcome to the ToDo app instructions")
```
Comment (edu-transcarent) [inline toDoList/cmd/main.go:32]:
```suggestion
	fmt.Println(`Welcome to the ToDo app instructions
	APIS: 
	
	`)
```

you can do something like that so you don't call printing method so many times

File: toDoList/cmd/main.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	fmt.Println("GET, POST, PUT, DELETE")
+	fmt.Println("/api/v1/events")
+}
+
+func CreateEvent(w http.ResponseWriter, r *http.Request) {
+	message := make(map[string]string)
+	var event Event
+	decoder := json.NewDecoder(r.Body)
+
+	w.Header().Set("Content-Type", "application/json")
```
Comment (edu-transcarent) [inline toDoList/cmd/main.go:49]: this line could be at the end of the method

File: toDoList/cmd/main.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	fmt.Println("Title: string")
+	fmt.Println("Description: string")
+
+	fmt.Println("This is our endpoint and the methods that the url can manage")
+	fmt.Println("GET, POST, PUT, DELETE")
+	fmt.Println("/api/v1/events")
+}
+
+func CreateEvent(w http.ResponseWriter, r *http.Request) {
+	message := make(map[string]string)
```
Comment (edu-transcarent) [inline toDoList/cmd/main.go:45]: Wouldn't be better to make this map a response struct ?

File: toDoList/cmd/main.go (hunk 1 above)
Comment (joaquincamara) [inline toDoList/cmd/main.go]: It is, its just to emulate a database

File: toDoList/cmd/main.go (hunk 2 above)
Comment (joaquincamara) [inline toDoList/cmd/main.go:34]: test for the main function?

File: toDoList/cmd/main.go (hunk 4 above)
Comment (joaquincamara) [inline toDoList/cmd/main.go]: line 49 ?

File: toDoList/cmd/main.go (hunk 1 above)
Comment (eduardo-villasenor-encora) [reply toDoList/cmd/main.go]: If that's the intention, probably the best idea would be to isolate a new db package

File: toDoList/cmd/main.go (hunk 4 above)
Comment (eduardo-villasenor-encora) [reply toDoList/cmd/main.go]: It doesn't have too much sense to start writing a response when you are still validating something

File: toDoList/cmd/main.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+		w.Write(response)
+	}
+	defer r.Body.Close()
+
+	if _, ok := DB[event.Id]; ok {
+		message["Exist"] = "Event already exist"
+		response, _ := json.Marshal(message)
+		w.WriteHeader(http.StatusFound)
+		w.Write(response)
+	} else {
```
Comment (eduardo-villasenor-encora) [inline toDoList/cmd/main.go]: you don't need the else

File: toDoList/cmd/main.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+	}
+	defer r.Body.Close()
+
+	if _, ok := DB[event.Id]; ok {
+		message["EditionSucces"] = "Edition succesful"
+		DB[event.Id] = event
+		response, _ := json.Marshal(message["EditionSucces"])
+		w.WriteHeader(http.StatusOK)
+		w.Write(response)
+	} else {
```
Comment (eduardo-villasenor-encora) [inline toDoList/cmd/main.go]: Don't need this else

File: toDoList/cmd/main.go [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+	}
+	defer r.Body.Close()
+
+	if _, ok := DB[event.Id]; ok {
+		message["DeleteSucces"] = "Delete succesful"
+		delete(DB, event.Id)
+		response, _ := json.Marshal(message["DeleteSucces"])
+		w.WriteHeader(http.StatusOK)
+		w.Write(response)
+	} else {
```
Comment (eduardo-villasenor-encora) [inline toDoList/cmd/main.go]: Don't need this else

File: toDoList/cmd/main.go [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if err := decoder.Decode(&event); err != nil {
+		message["error"] = "Invalid request"
+		response, _ := json.Marshal(message["error"])
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write(response)
+	}
+	defer r.Body.Close()
```
Comment (eduardo-villasenor-encora) [inline toDoList/cmd/main.go]: question , are you going to log the error if you can't close this body ?

File: toDoList/cmd/main.go (hunk 2 above)
Comment (eduardo-villasenor-encora) [reply toDoList/cmd/main.go:34]: mmmm Nop , you should at least test that the route works

File: toDoList/cmd/main.go (hunk 9 above)
Comment (joaquincamara) [reply toDoList/cmd/main.go]: Did not consider that, yes, sounds like a good idea

**Pull request merged**